test/org/sagebionetworks/bridge/services/EmailVerificationServiceTest.java

```diff
@@ -0,0 +1,125 @@
+                
+        Map<String,IdentityVerificationAttributes> map = Maps.newHashMap();
+        map.put(EMAIL_ADDRESS, attributes);
+        when(result.getVerificationAttributes()).thenReturn(map);
+        when(attributes.getVerificationStatus()).thenReturn(status); // aka unverified
+        when(sesClient.getIdentityVerificationAttributes(any())).thenReturn(result);
+    }
+    
+    @Test
+    public void verifiedEmailTakesNoAction() {
+        mockSession("Success");
+
+        EmailVerificationStatus status = service.verifyEmailAddress(EMAIL_ADDRESS);
+        
+        assertEquals(EmailVerificationStatus.VERIFIED, status);
+        verify(sesClient, never()).verifyEmailIdentity(any());
+        verify(sesClient).getIdentityVerificationAttributes(getCaptor.capture());
+        assertEquals(EMAIL_ADDRESS, getCaptor.getValue().getIdentities().get(0));
+    }
+    
+    @Test
+    public void ifUnverifiedAttemptsToResendVerification() {
+        mockSession("Failure");
+
+        EmailVerificationStatus status = service.verifyEmailAddress(EMAIL_ADDRESS);
+        
+        assertEquals(EmailVerificationStatus.PENDING, status);
+        verify(sesClient).verifyEmailIdentity(any());
+        verify(sesClient).getIdentityVerificationAttributes(getCaptor.capture());
+        assertEquals(EMAIL_ADDRESS, getCaptor.getValue().getIdentities().get(0));
+    }
+    
+    @Test
+    public void emailDoesntExistRequestVerification() {
+        ArgumentCaptor<VerifyEmailIdentityRequest> verifyCaptor = ArgumentCaptor.forClass(VerifyEmailIdentityRequest.class);
+        mockSession(null);
+        
+        EmailVerificationStatus status = service.verifyEmailAddress(EMAIL_ADDRESS);
+        
+        assertEquals(EmailVerificationStatus.PENDING, status);
+        verify(sesClient).verifyEmailIdentity(verifyCaptor.capture());
+        verify(sesClient).getIdentityVerificationAttributes(getCaptor.capture());
+        assertEquals(EMAIL_ADDRESS, verifyCaptor.getValue().getEmailAddress());
+        assertEquals(EMAIL_ADDRESS, getCaptor.getValue().getIdentities().get(0));
+    }
+
+    @Test
+    public void canResendRegardlessOfStatus() {
+        ArgumentCaptor<VerifyEmailIdentityRequest> verifyCaptor = ArgumentCaptor.forClass(VerifyEmailIdentityRequest.class);
+        mockSession("Success");
+        
+        EmailVerificationStatus status = service.sendVerifyEmailRequest(EMAIL_ADDRESS);
+        
+        assertEquals(EmailVerificationStatus.PENDING, status);
+        verify(sesClient).verifyEmailIdentity(verifyCaptor.capture());
+        assertEquals(EMAIL_ADDRESS, verifyCaptor.getValue().getEmailAddress());
+    }
+    
+    @Test
+    public void getEmailStatus() {
+        mockSession("Success");
+        
+        EmailVerificationStatus status = service.getEmailStatus(EMAIL_ADDRESS);
+        
+        verify(sesClient).getIdentityVerificationAttributes(any());
+        assertEquals(EmailVerificationStatus.VERIFIED, status);
+    }
```

Comment:
As set up, studies can't be saved without a support email and that's what we pass in. So there's no validation of a submitted value here because no value is submitted from the user. There's emailDoesntExistRequestVerification() which tests what happens if SES has not seen the email before (you get back a null value for value in the attribute map).